backend/src/main/java/botobo/core/domain/tag/TagRepository.java

```diff
@@ -3,16 +3,22 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
-import org.springframework.stereotype.Repository;
 
 import java.util.List;
 import java.util.Optional;
 
-@Repository
 public interface TagRepository extends JpaRepository<Tag, Long> {
 
     Optional<Tag> findByTagName(TagName tagName);
 
     @Query("select t from Tag t where t.tagName.value like %:keyword%")
     List<Tag> findAllTagContaining(@Param("keyword") String keyword);
+
+    @Query(value = "SELECT * FROM tag t WHERE t.NAME LIKE %:keyword% LIMIT 10", nativeQuery = true)
+    List<Tag> findByKeyword(@Param("keyword") String keyword);
+
+    @Query("select t from Tag t join fetch t.workbookTags wt " +
+            "where t.id = wt.tag.id " +
+            "and lower(wt.workbook.name) like %:workbookName%")
+    List<Tag> findAllByWorkbookName(String workbookName);
```

Comment:
```suggestion
    Set<Tag> findAllByWorkbookName(String workbookName);
```
저도 이것저것 만져보며 발견했는데 List를 new HashSet으로 감싸지 않아도 이렇게 Set으로 받아오는게 가능하네요 😮

Reply:
파라미터명이랑 변수랑 같아서 @Param 없어도 되는걸로 알고있어유!!